spec/lib/card/format_spec.rb

```diff
@@ -3,33 +3,80 @@
-      expect(@format.show_view?( :menu, :hide=>'menu, paging', :default_visibility=>:show )).to be_falsey
-      expect(@format.show_view?( :menu, :show=>'menu', :optional_menu=>:hide              )).to be_truthy      
+      expect(format.show_view?( :menu, :show=>'menu', :default_visibility=>:hide         )).to be_truthy
+      expect(format.show_view?( :menu, :hide=>'menu, paging', :default_visibility=>:show )).to be_falsey
+      expect(format.show_view?( :menu, :show=>'menu', :optional_menu=>:hide              )).to be_truthy      
     end
     
     it "should handle hard developer overrides" do
-      expect(@format.show_view?( :menu, :optional_menu=>:always, :hide=>'menu' )).to be_truthy
-      expect(@format.show_view?( :menu, :optional_menu=>:never,  :show=>'menu' )).to be_falsey
+      expect(format.show_view?( :menu, :optional_menu=>:always, :hide=>'menu' )).to be_truthy
+      expect(format.show_view?( :menu, :optional_menu=>:never,  :show=>'menu' )).to be_falsey
     end
     
   end
 
+  describe 'format helpers and link building' do
+    before :each do
+      # should be a way to get these defaults in these tests
+      Card::Env[:host] = '//test.host'
+      Card::Env[:protocol] = 'http:'
+    end
+
+    let(:card) { Card['Home'] }
+    let(:text_format) { card.format(:format=>:text) }
+    let(:html_format) { card.format }
+    let(:url_text1) { 'with external free link http://localhost:2020/path?cgi=foo&bar=baz' }
+    let(:url_text2) { 'with external in link syntax: [[http://brain.org/Home|extra]]' }
+    let(:url_text3) { 'with internal lik [[A]]' }
+    let(:url_text4) { 'with internal lik [[Home|display text]]' }
+    let(:url_text5) { 'external with port: http://localhost:2020/path?cgi=foo+bar=baz after '}
+
+    it "should format links" do
+      cobj = Card::Content.new url_text1, text_format
+      expect(cobj.to_s).to eq url_text1
+      cobj = Card::Content.new url_text2, text_format
+      expect(cobj.to_s).to eq url_text2
+      cobj = Card::Content.new url_text3, text_format
+      expect(cobj.to_s).to eq url_text3
+      cobj = Card::Content.new url_text4, text_format
+      expect(cobj.to_s).to eq url_text4
+      cobj = Card::Content.new url_text5, text_format
+      expect(cobj.to_s).to eq url_text5
+    end
+
+    it "should format html links" do
+      cobj = Card::Content.new url_text1, html_format
+      expect(cobj.to_s).to eq "with external free link <a class=\"external-link\" href=\"http://localhost:2020/path?cgi=foo&amp;bar=baz\">http://localhost:2020/path?cgi=foo&amp;bar=baz</a>"
+      cobj = Card::Content.new url_text2+url_text3+url_text4, html_format
+      expect(cobj.to_s).to eq url_text2+url_text3+url_text4
+      cobj = Card::Content.new url_text5, html_format
+      expect(cobj.to_s).to eq "external with port: <a class=\"external-link\" href=\"http://localhost:2020/path?cgi=foo+bar=baz\">http://localhost:2020/path?cgi=foo+bar=baz</a> after "
```

Comment:
Re the first link above, it's clearly good to encode the html in the "text" part of the link. In the href attribute, it seems odd to be altering a url, which creates a serious risk of double encoding.  Can't we just leave it unaltered there?

Re the second link, this is just a screwy CGI query, although in this case at least it starts off that way.